### ISSUE TITLE
Why did you use dynamic bucketing in Canary training and didnt in stt eu asr training

### ISSUE THREAD

**Issue by SatenHarutyunyan** [NVIDIA/NeMo#9762]
I see from the paper that you used dynamic bucketing - a way to reducing padding amount, and in the stt-eu config I see that    
```
bucketing_strategy: synced_randomized
bucketing_batch_size: null
 
```
   I assume this null means not doing that. 
    I wonder why?

**Comment by pzelasko** [comment]
Hi @SatenHarutyunyan, I don't know about `stt-eu` recipe but I can offer some insights. Briefly before we released Canary-1B, we added [lhotse](https://github.com/lhotse-speech/lhotse) as an alternative dataloading backend for NeMo. The original NeMo dataloading implementation only allows you to do "static" bucketing where you have to manually partition your data into buckets and assign a batch size to each. With lhotse dataloading you don't have to do any of that; instead, you set the configuration for the number of buckets and (optionally) the bucket duration bins, and the total audio duration to be seen in a mini-batch. This results in dynamic bucketing and dynamic batch sizes. 

So in short, dynamic bucketing (and other lhotse features) were only available in NeMo shortly before the release of Canary-1B.

**Comment by SatenHarutyunyan** [comment]
Thank you for the answer, thats very helpful.

I was asking about this recipe. https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/stt_multilingual_fastconformer_hybrid_large_pc_blend_eu

If you use Ihotse for dynamic bucketing, would that be seen in the config.yaml file that yu released?
Does dynamic bucketing help only for training speed or it helps the model training/convergence, better results in simple words?

**Comment by pzelasko** [comment]
> If you use Ihotse for dynamic bucketing, would that be seen in the config.yaml file that you released?

Yes, unless the config was minimized before release.

> Does dynamic bucketing help only for training speed or it helps the model training/convergence, better results in simple words?

It helps with speeding up the convergence due to larger batch sizes. You might be interested in our very recent work that allows to tune the batch size to the maximum handled by the GPU for a given model, we've seen solid speedups (same result even 2x faster as measured by wall time on the same number on GPUs).

#9763
